pydantic/dataclasses.py

```diff
@@ -233,9 +233,10 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
             if config_wrapper.frozen:
                 # It's not recommended to define both, as the setting from the dataclass decorator will take priority.
                 warn(
-                    f"`frozen` is set via both the 'dataclass' decorator and `config` for dataclass {cls.__name__}."
+                    f'`frozen` is set via both the `dataclass` decorator and `config` for dataclass {cls.__name__}.'
                     'This is not recommended. The `frozen` specification on `dataclass` will take priority.',
-                    UserWarning,
+                    category=UserWarning,
+                    stacklevel=3,
```

Comment:
```suggestion
                    stacklevel=2,
```

Turns out I was wrong, `stacklevel=2` is the right one to have the warning emitted at the correct place :sweat_smile: